gui/app/services/effectHelperService.js

```diff
@@ -102,6 +102,79 @@
           };
           break;
 
+        case EffectType.CHANGE_SCENE:
+          controller = ($scope, groupsService, boardService) => {
+
+            // Make the Change Scene option user friendly text.
+            // EX: effect.reset = false should show as "Reset Scenes" in the ui.
+            $scope.ufChangeSceneOption = function(){
+              if($scope.effect.reset){
+                return "Reset Scenes"
+              } else if($scope.effect.reset === false){
+                return "Change Scenes"
+              } else {
+                return "Pick One"
+              }
+            }
+
+            // Load up viewer groups if they haven't been already.
+            // Leaving this out causes no groups to load unless you first visit the groups panel.
+            groupsService.loadViewerGroups();
+
+            // Get viewer groups and push group name to scope.
+            // This is for loading up all user group checkboxes.
+            $scope.viewerGroups = [];
+            var groups = groupsService.getViewerGroups();
+            for (group of groups){
+              $scope.viewerGroups.push(group.groupName);
+            }
+
+            // This is run each time a group checkbox is clicked or unclicked.
+            // This will build an array of currently selected groups to be saved to JSON.
+            $scope.groupArray = function(group){
+              if($scope.effect.groups !== undefined){
```

Comment:
I recently looked this up and learned that it's best practice to do `variable != null` since null and undefined are two different things. Doing `!== undefined` only checks for undefined but doing `!= null` will convert "undefined" to null so you cover both possibilities. While it's unlikely that this will be null, I don't think it hurts to be safe.